### ISSUE TITLE
I don’t know if someone is like me. Using this plugin in unraid 6.12 will cause the SSH service to be closed after booting and cannot be opened

### ISSUE THREAD

**Issue by how1ewu** [dkaser/unraid-tailscale#57]
I don’t know if someone is like me. Using this plugin in unraid 6.12 will cause the SSH service to be closed after booting and cannot be opened

**Comment by dkaser** [comment]
Reported on forums via https://forums.unraid.net/topic/136889-plugin-tailscale/?do=findComment&comment=1272477

Waiting for diagnostics from user.

**Comment by iamwoz** [comment]
yes I have exactly this problem, removing the plugin restores SSH functionality - I see @how1ewu diagnostics are on the forum now, do you want mine as well?

**Comment by dkaser** [comment]
I think I see what's happening. Out of curiosity, do you have a /boot/config/plugins/tailscale/tailscale.cfg file?

**Comment by how1ewu** [comment]
> /boot/config/plugins/tailscale/tailscale.cfg

ACCEPT_ROUTES="1"
ACCEPT_DNS="0"
INCLUDE_INTERFACE="1"
SYSCTL_IP_FORWARD="1"

**Comment by how1ewu** [comment]
> I think I see what's happening. Out of curiosity, do you have a /boot/config/plugins/tailscale/tailscale.cfg file?

ACCEPT_ROUTES="1"
ACCEPT_DNS="0"
INCLUDE_INTERFACE="1"
SYSCTL_IP_FORWARD="1"

**Comment by dkaser** [comment]
OK, that's not what I thought was happening then.

I think I remember seeing a post in one of the release threads about a similar issue related to disabling IPv6. Do you have anything in your /boot/config/go file? I'll go try to look up the post.

**Comment by how1ewu** [comment]
> /boot/config/go
I have a frp intranet penetration service and a cloudflared service
![image](https://github.com/dkaser/unraid-tailscale/assets/98788152/4eabea15-079f-478d-b408-ed86dcf7a740)

**Comment by how1ewu** [comment]
> OK, that's not what I thought was happening then.
> 
> I think I remember seeing a post in one of the release threads about a similar issue related to disabling IPv6. Do you have anything in your /boot/config/go file? I'll go try to look up the post.

cloudflared service is disabled by default

**Comment by dkaser** [comment]
@how1ewu What do you get if you run `/etc/rc.d/rc.sshd restart` in the web console?

@iamwoz Diagnostics would be great, my first thought has been proven wrong already

**Comment by how1ewu** [comment]
> @how1ewu What do you get if you run `/etc/rc.d/rc.sshd restart` in the web console?
> 
> @iamwoz Diagnostics would be great, my first thought has been proven wrong already

Nothing happens, the sshd service still fails to start, there are no errors or log output on the console

**Comment by dkaser** [comment]
That's very odd. It seems like Unraid is detecting the Tailscale IP correctly from the diagnostics, so I'm not certain why SSH wouldn't be starting.

This seems almost like something weird with the new Unraid "extra interfaces" setup... the plugin seems to have done everything correctly from what I can see... tailscaled is starting fine and adding it's interface to the network-extra.cfg file just like it should.

Can you post a copy of your /etc/ssh/sshd_config file? That might help me pin down what's happening.

**Comment by how1ewu** [comment]
> That's very odd. It seems like Unraid is detecting the Tailscale IP correctly from the diagnostics, so I'm not certain why SSH wouldn't be starting.
> 
> This seems almost like something weird with the new Unraid "extra interfaces" setup... the plugin seems to have done everything correctly from what I can see... tailscaled is starting fine and adding it's interface to the network-extra.cfg file just like it should.
> 
> Can you post a copy of your /etc/ssh/sshd_config file? That might help me pin down what's happening.

root@howienas:~# cat  /etc/ssh/sshd_config
#       $OpenBSD: sshd_config,v 1.104 2021/07/02 05:11:21 dtucker Exp $

# This is the sshd server system-wide configuration file.  See
# sshd_config(5) for more information.

# This sshd was compiled with PATH=/usr/local/sbin:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin

# The strategy used for options in the default sshd_config shipped with
# OpenSSH is to specify options with their default value where
# possible, but leave them commented.  Uncommented options override the
# default value.

Port 22
AddressFamily inet
ListenAddress 192.168.1.66
ListenAddress 100.69.69.16
ListenAddress fd7a:115c:a1e0:ab12:4843:cd96:6245:4510

#HostKey /etc/ssh/ssh_host_rsa_key
#HostKey /etc/ssh/ssh_host_ecdsa_key
#HostKey /etc/ssh/ssh_host_ed25519_key

# Ciphers and keying
#RekeyLimit default none

# Logging
#SyslogFacility AUTH
#LogLevel INFO
# limetech - lets be verbose
LogLevel VERBOSE

# Authentication:
# limetech - we need to allow root
AllowUsers root wsh
PermitRootLogin yes

#LoginGraceTime 2m
PermitRootLogin yes
#StrictModes yes
MaxAuthTries 100
#MaxSessions 10

#PubkeyAuthentication yes

# The default is to check both .ssh/authorized_keys and .ssh/authorized_keys2
# but this is overridden so installations will only check .ssh/authorized_keys
AuthorizedKeysFile      .ssh/authorized_keys

#AuthorizedPrincipalsFile none

#AuthorizedKeysCommand none
#AuthorizedKeysCommandUser nobody

# For this to work you will also need host keys in /etc/ssh/ssh_known_hosts
#HostbasedAuthentication no
# Change to yes if you don't trust ~/.ssh/known_hosts for
# HostbasedAuthentication
#IgnoreUserKnownHosts no
# Don't read the user's ~/.rhosts and ~/.shosts files
#IgnoreRhosts yes

# To disable tunneled clear text passwords, change to no here!
PasswordAuthentication yes
PermitEmptyPasswords no

# Change to no to disable s/key passwords
#KbdInteractiveAuthentication yes

# Kerberos options
#KerberosAuthentication no
#KerberosOrLocalPasswd yes
#KerberosTicketCleanup yes
#KerberosGetAFSToken no

# GSSAPI options
#GSSAPIAuthentication no
#GSSAPICleanupCredentials yes

# Set this to 'yes' to enable PAM authentication, account processing,
# and session processing. If this is enabled, PAM authentication will
# be allowed through the KbdInteractiveAuthentication and
# PasswordAuthentication.  Depending on your PAM configuration,
# PAM authentication via KbdInteractiveAuthentication may bypass
# the setting of "PermitRootLogin without-password".
# If you just want the PAM account and session checks to run without
# PAM authentication, then enable this but set PasswordAuthentication
# and KbdInteractiveAuthentication to 'no'.
UsePAM yes

#AllowAgentForwarding yes
#AllowTcpForwarding yes
# limeteh - disallow forwarding but see below
AllowTcpForwarding no                  
GatewayPorts no
#X11Forwarding no
#X11DisplayOffset 10
#X11UseLocalhost yes
#PermitTTY yes
#PrintMotd yes
#PrintLastLog yes
#TCPKeepAlive yes
#PermitUserEnvironment no
#Compression delayed
#ClientAliveInterval 0
#ClientAliveCountMax 3
#UseDNS no
#PidFile /var/run/sshd.pid
#MaxStartups 10:30:100
#PermitTunnel no
#ChrootDirectory none
#VersionAddendum none

# no default banner path
#Banner none

# override default of no subsystems
Subsystem       sftp    /usr/libexec/sftp-server

# Example of overriding settings on a per-user basis
#Match User anoncvs
#       X11Forwarding no
#       AllowTcpForwarding no
#       PermitTTY no
#       ForceCommand cvs server

# limetech - permit only root SSH tunneling
Match Group root
AllowTcpForwarding yes

**Comment by how1ewu** [comment]
> That's very odd. It seems like Unraid is detecting the Tailscale IP correctly from the diagnostics, so I'm not certain why SSH wouldn't be starting.
> 
> This seems almost like something weird with the new Unraid "extra interfaces" setup... the plugin seems to have done everything correctly from what I can see... tailscaled is starting fine and adding it's interface to the network-extra.cfg file just like it should.
> 
> Can you post a copy of your /etc/ssh/sshd_config file? That might help me pin down what's happening.

192.168.1.66 is my local address
100.69.69.16 is the tailscale address
```shell
root@howienas:~# cat  /etc/ssh/sshd_config
#       $OpenBSD: sshd_config,v 1.104 2021/07/02 05:11:21 dtucker Exp $

# This is the sshd server system-wide configuration file.  See
# sshd_config(5) for more information.

# This sshd was compiled with PATH=/usr/local/sbin:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin

# The strategy used for options in the default sshd_config shipped with
# OpenSSH is to specify options with their default value where
# possible, but leave them commented.  Uncommented options override the
# default value.

Port 22
AddressFamily inet
ListenAddress 192.168.1.66
ListenAddress 100.69.69.16
ListenAddress fd7a:115c:a1e0:ab12:4843:cd96:6245:4510

#HostKey /etc/ssh/ssh_host_rsa_key
#HostKey /etc/ssh/ssh_host_ecdsa_key
#HostKey /etc/ssh/ssh_host_ed25519_key

# Ciphers and keying
#RekeyLimit default none

# Logging
#SyslogFacility AUTH
#LogLevel INFO
# limetech - lets be verbose
LogLevel VERBOSE

# Authentication:
# limetech - we need to allow root
AllowUsers root wsh
PermitRootLogin yes

#LoginGraceTime 2m
PermitRootLogin yes
#StrictModes yes
MaxAuthTries 100
#MaxSessions 10

#PubkeyAuthentication yes

# The default is to check both .ssh/authorized_keys and .ssh/authorized_keys2
# but this is overridden so installations will only check .ssh/authorized_keys
AuthorizedKeysFile      .ssh/authorized_keys

#AuthorizedPrincipalsFile none

#AuthorizedKeysCommand none
#AuthorizedKeysCommandUser nobody

# For this to work you will also need host keys in /etc/ssh/ssh_known_hosts
#HostbasedAuthentication no
# Change to yes if you don't trust ~/.ssh/known_hosts for
# HostbasedAuthentication
#IgnoreUserKnownHosts no
# Don't read the user's ~/.rhosts and ~/.shosts files
#IgnoreRhosts yes

# To disable tunneled clear text passwords, change to no here!
PasswordAuthentication yes
PermitEmptyPasswords no

# Change to no to disable s/key passwords
#KbdInteractiveAuthentication yes

# Kerberos options
#KerberosAuthentication no
#KerberosOrLocalPasswd yes
#KerberosTicketCleanup yes
#KerberosGetAFSToken no

# GSSAPI options
#GSSAPIAuthentication no
#GSSAPICleanupCredentials yes

# Set this to 'yes' to enable PAM authentication, account processing,
# and session processing. If this is enabled, PAM authentication will
# be allowed through the KbdInteractiveAuthentication and
# PasswordAuthentication.  Depending on your PAM configuration,
# PAM authentication via KbdInteractiveAuthentication may bypass
# the setting of "PermitRootLogin without-password".
# If you just want the PAM account and session checks to run without
# PAM authentication, then enable this but set PasswordAuthentication
# and KbdInteractiveAuthentication to 'no'.
UsePAM yes

#AllowAgentForwarding yes
#AllowTcpForwarding yes
# limeteh - disallow forwarding but see below
AllowTcpForwarding no                  
GatewayPorts no
#X11Forwarding no
#X11DisplayOffset 10
#X11UseLocalhost yes
#PermitTTY yes
#PrintMotd yes
#PrintLastLog yes
#TCPKeepAlive yes
#PermitUserEnvironment no
#Compression delayed
#ClientAliveInterval 0
#ClientAliveCountMax 3
#UseDNS no
#PidFile /var/run/sshd.pid
#MaxStartups 10:30:100
#PermitTunnel no
#ChrootDirectory none
#VersionAddendum none

# no default banner path
#Banner none

# override default of no subsystems
Subsystem       sftp    /usr/libexec/sftp-server

# Example of overriding settings on a per-user basis
#Match User anoncvs
#       X11Forwarding no
#       AllowTcpForwarding no
#       PermitTTY no
#       ForceCommand cvs server

# limetech - permit only root SSH tunneling
Match Group root
AllowTcpForwarding yes
```

**Comment by dkaser** [comment]
OK, that's very helpful. The problem is this:

`AddressFamily inet`

That sets ssh to accept IPv4 addresses only. Let me look at the Unraid script that builds the config, then I can get a bug report submitted on their side to fix it.

**Comment by how1ewu** [comment]
> OK, that's very helpful. The problem is this:
> 
> `AddressFamily inet`
> 
> That sets ssh to accept IPv4 addresses only. Let me look at the Unraid script that builds the config, then I can get a bug report submitted on their side to fix it.
Thank you for your efforts, hope to fix the problem as soon as possible

**Comment by how1ewu** [comment]
> OK, that's very helpful. The problem is this:
> 
> `AddressFamily inet`
> 
> That sets ssh to accept IPv4 addresses only. Let me look at the Unraid script that builds the config, then I can get a bug report submitted on their side to fix it.

Thank you for your efforts, hope to fix the problem as soon as possible

**Comment by how1ewu** [comment]
> OK, that's very helpful. The problem is this:
> 
> `AddressFamily inet`
> 
> That sets ssh to accept IPv4 addresses only. Let me look at the Unraid script that builds the config, then I can get a bug report submitted on their side to fix it.

Does this mean that I can solve the problem by turning off unraid ipv6?
In addition, my logs often report the following errors, but I don’t feel the impact on the use. Can I ignore them?
```shell
tailscaled: 2023/06/17 10:47:26 Received error: PollNetMap: unexpected EOF
tailscaled: 2023/06/17 10:45:08 health("overall"): error: not connected to home DERP region 20
tailscaled: 2023/06/17 10:42:24 Received error: PollNetMap: Post "https://controlplane.tailscale.com/machine/map": read tcp 192.168.1.66:44978->18.156.90.224:80: use of closed network connection
```

**Comment by dkaser** [comment]
Those errors are fairly normal.

Disabling IPv6 on Unraid will probably not help matters -- actually, what would fix the problem would be getting a valid IPv6 address for br0. I'm looking to see if there's a workaround that would prevent Tailscale from registering its IPv6 address, but I'm not super hopeful on that... we'll probably have to wait for the Unraid scripts to get fixed.

**Comment by dkaser** [comment]
Bug report posted: https://forums.unraid.net/bug-reports/stable-releases/612-ssh-fails-to-start-when-normal-interfaces-only-have-ipv4-but-extra-interfaces-have-ipv6-r2470/

I don't see any way to easily work around this on the Tailscale side. This should be an easy fix for Unraid, though, so it will hopefully make it to their next release.

**Comment by how1ewu** [comment]
> Those errors are fairly normal.
> 
> Disabling IPv6 on Unraid will probably not help matters -- actually, what would fix the problem would be getting a valid IPv6 address for br0. I'm looking to see if there's a workaround that would prevent Tailscale from registering its IPv6 address, but I'm not super hopeful on that... we'll probably have to wait for the Unraid scripts to get fixed.

I turned off IPV4+IPV6 in the unraid network settings and kept only IPV4, now I can use SSH and tailscale normally, this may be a solution

**Comment by how1ewu** [comment]
![](https://s2.loli.net/2023/06/17/1oZLWtkQzn8SPlc.png)
![](https://s2.loli.net/2023/06/17/kJ2U9wngCGBdDWH.png)

**Comment by dkaser** [comment]
That surprises me, but I'm glad it works 👍

**Comment by how1ewu** [comment]
> That surprises me, but I'm glad it works 👍

Yes, I think this is a temporary solution, but in the end we still need to wait for the unraid official to fix it, which is a problem that seriously affects the use

**Comment by ericswpark** [comment]
@dkaser I'm running into the same issue. As a temporary workaround, could I edit the `/etc/rc.d/rc.library.source` file and replace `inet` with `any`? I assume all files in `rc.d` are replaced when the server reboots, so if that is fine it should tide me over until the next stable release.

I tried the IPv4 only trick that @how1ewu wrote about, but found that my server was already configured to IPv4 only. I tried flipping between IPv4+6 and IPv4 again to see if that would temporarily fix the problem but it didn't for my server.
